packages/flutter/test/material/navigation_rail_test.dart

```diff
@@ -3291,6 +3291,53 @@ void main() {
     tester.pumpAndSettle();
   });
 
+  testWidgetsWithLeakTracking("Destination's label with the right opacity while disabled", (WidgetTester tester) async {
+    await _pumpNavigationRail(
+      tester,
+      navigationRail: NavigationRail(
+        selectedIndex: 0,
+        destinations: const <NavigationRailDestination>[
+          NavigationRailDestination(
+            icon: Icon(Icons.favorite_border),
+            selectedIcon: Icon(Icons.favorite),
+            label: Text('Abc'),
+          ),
+          NavigationRailDestination(
+            icon: Icon(Icons.bookmark_border),
+            selectedIcon: Icon(Icons.bookmark),
+            label: Text('Bcd'),
+            disabled: true,
+          ),
+        ],
+        onDestinationSelected: (int index) {},
+        labelType: NavigationRailLabelType.all,
+      ),
+    );
+
+    await tester.pump();
+
+    final double? abcLabelOpacity = tester.widget<DefaultTextStyle>(
+      find.ancestor(
+        of: find.text('Abc'),
+        matching: find.byType(DefaultTextStyle),
+      ).first,
+    ).style.color?.opacity;
+    final double? bcdLabelOpacity = tester.widget<DefaultTextStyle>(
+      find.ancestor(
+        of: find.text('Bcd'),
+        matching: find.byType(DefaultTextStyle),
+      ).first,
+    ).style.color?.opacity;
+
+    expect(abcLabelOpacity, 1.0);
+
+    if (bcdLabelOpacity == 1.0) {
```

Comment:
It shouldn't be necessary to soften this test to `!= 1` since we're explicitly setting the opacity to 0.38.